provider/rosa_operator_roles_data_source.go

```diff
@@ -0,0 +1,229 @@
+		Attributes: map[string]tfsdk.Attribute{
+			"cluster_id": {
+				Description: "Cluster id.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"operator_role_prefix": {
+				Description: "Operator role prefix.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"account_role_prefix": {
+				Description: "Account role prefix.",
+				Type:        types.StringType,
+				Optional:    true,
+			},
+			"operator_iam_roles": {
+				Description: "Operator IAM Roles.",
+				Attributes: tfsdk.ListNestedAttributes(
+					t.itemAttributes(),
+					tfsdk.ListNestedAttributesOptions{},
+				),
+				Computed: true,
+			},
+		},
+	}
+	return
+}
+
+func (t *RosaOperatorRolesDataSourceType) itemAttributes() map[string]tfsdk.Attribute {
+	return map[string]tfsdk.Attribute{
+		"operator_name": {
+			Description: "Operator Name",
+			Type:        types.StringType,
+			Computed:    true,
+		},
+		"operator_namespace": {
+			Description: "Kubernetes Namespace",
+			Type:        types.StringType,
+			Computed:    true,
+		},
+		"role_arn": {
+			Description: "AWS Role ARN",
+			Type:        types.StringType,
+			Computed:    true,
+		},
+		"role_name": {
+			Description: "policy name",
+			Type:        types.StringType,
+			Computed:    true,
+		},
+		"policy_name": {
+			Description: "policy name",
+			Type:        types.StringType,
+			Computed:    true,
+		},
+		"service_accounts": {
+			Description: "service accounts",
+			Type: types.ListType{
+				ElemType: types.StringType,
+			},
+			Computed: true,
+		},
+	}
+}
+
+func (t *RosaOperatorRolesDataSourceType) NewDataSource(ctx context.Context,
+	p tfsdk.Provider) (result tfsdk.DataSource, diags diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation:
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
+	clustersClient := parent.connection.ClustersMgmt().V1().Clusters()
+	awsInquiries := parent.connection.ClustersMgmt().V1().AWSInquiries()
+
+	// Create the resource:
+	result = &RosaOperatorRolesDataSource{
+		logger:         parent.logger,
+		clustersClient: clustersClient,
+		awsInquiries:   awsInquiries,
+	}
+	return
+}
+
+func (t *RosaOperatorRolesDataSource) Read(ctx context.Context, request tfsdk.ReadDataSourceRequest,
+	response *tfsdk.ReadDataSourceResponse) {
+	// Get the state:
+	state := &RosaOperatorRolesState{}
+	diags := request.Config.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	stsOperatorRolesList, err := t.awsInquiries.STSCredentialRequests().List().Send()
+	if err != nil {
+		t.logger.Error(ctx, "Failed to get operator list")
+		return
+	}
+
+	stsOperatorMap := make(map[string]*cmv1.STSOperator)
+	stsOperatorRolesList.Items().Each(func(stsCredentialRequest *cmv1.STSCredentialRequest) bool {
+		t.logger.Debug(ctx, "Operator name: %s, namespace %s, service account %s",
+			stsCredentialRequest.Operator().Name(),
```

Comment:
If not mistaken, this response also has a min openshift version so we can check if the cluster actually needs a specific operator

Reply:
I plan to check it as part of that ticket: https://issues.redhat.com/browse/SDA-6953 
I was afraid that it will be too much changes in one PR

Reply:
I added a todo comment